README.md

```diff
@@ -30,165 +32,557 @@ The two guiding principles are:
 
-In addition to the core presets that [we mentioned earlier](#presets) these following presets are available to help you with the struggles of universal rendering, deployments, service workers and such.
+**`package.json`**
 
-#### `hops-react`
+```json
+{
+  "hops": {
+    "foo": {
+      "bar": "baz"
+    },
+    "qux": "<foo.bar>"
+  }
+}
+```
 
-This preset enables usage of React.js with JSX, React-Helmet and React-Router support in Hops applications.
+#### Environment variables
 
-Install it to your project:
+Hops has a concept of "universal environment variables", which are environment variables that are evaluated at _runtime_ instead of at _build time_.
+
+Usually when you reference `process.env.FOO` in your code, webpack will replace that expression with the value that the environment variable `FOO` had at build time. This is also true for Hops applications.
+
+**But:** Since Hops renders applications on the server- and client-side, we can reference environment variables in the settings and pass them on to the client and therefore allow users to have environment variables that are evaluated at runtime.
+
+**Caution:** Be careful not to expose secrets to the client side, because all environment variables that you reference this way in the settings will be visible in the rendered HTML.
 
-```bash
-$ yarn add hops-react@next react-dom react-helmet
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "myApiUrl": "[MY_API_URL]"
+  }
+}
 ```
 
-And write your main entry file (which should either be named `index.js` or should be referenced in your `package.json`s `main` field):
+The above example demonstrates how you can pass the value of the environment variable named `MY_API_URL` to Hops and later on you can access it in your application through its key like this:
+
+**`src/app.js`**
+
+```javascript
+import { render, withConfig } from 'hops';
+import React from 'react';
+
+const Config = withConfig(({ config }) => <div>{config.myApiUrl}</div>);
+
+export default render(<Config />);
+```
+
+This allows you to have just one build artifact that you can use in different environments (staging, production, etc).
+
+**Tip:** You can use Hops's configuration mechanism to provide custom values to your application. As seen in the environment variable example above, you can specify arbitrary keys in your settings and access them in your React application.
+
+**Tip:** Hops ships with integrated support for [`dotenv`](https://github.com/motdotla/dotenv), which means that it will try to read a `.env` file fron your application root directory and load its values as environment variables.
+
+### Options
+
+You can provide options to your application by passing an options hash (a plain JavaScript object) as the second argument to the [`render()`](#render) function in order to provide runtime configuration.
+
+**`src/app.js`**
+
+```javascript
+export default render(<MyApp />, { router: { basename: '/foo' } });
```

Comment:
I'm not sure `basename` is the best render option example as it's already ought to be set via config.basePath. I'm afraid it could make distinguishing config and render options even more confusing.